### DISCUSSION TITLE
Distinguish between requests coming from internal network and external networks (Internet)

### DISCUSSION THREAD

**Discussion by yasitha-pf** [istio/istio#44659] [Q&A]
Hi,

I am looking for a solution to differentiate between requests that are coming from internal networks and external networks.

For instance, I have an application called **A** that is called by another **internal** application called **B** AND from the **Internet** as well. I would like to disable authentication for requests originating from the internal network (e.g., 10.0.0.0/16 CIDR block). However, I noticed that when the request is from the Internet, the X-Forwarded-For header is logged, but I don't see this header for requests coming from the internal network.

Could you suggest the best way to differentiate between requests coming from internal and external networks?

**Comment by ramaraochavali** [comment]
x-envoy-internal will be set for all internal requests

**Reply by yasitha-pf** [reply]
I'm only receiving the following log.

```
{"start_time":"2023-05-01T18:08:44.883Z","bytes_received":0,"authority":"my-app-ae-local","duration":28,"bytes_sent":60,"request_id":"4896592c-6429-93df-938e-aecc8e4d3e42","path":"/agents/61071/leaves/summary?my_agent_id=215672","response_code":403,"upstream_cluster":"inbound|80||","requested_server_name":null,"upstream_local_address":"127.0.0.6:60297","method":"GET","route_name":"default","downstream_local_address":"10.23.94.242:80","x_forwarded_for":null,"protocol":"HTTP/1.1","response_flags":"-","downstream_remote_address":"10.102.43.50:50472","user_agent":"curl/8.0.1","response_code_details":"via_upstream","upstream_host":"10.23.94.242::80","upstream_transport_failure_reason":null,"connection_termination_details":null,"upstream_service_time":"28"}
```

**Reply by ramaraochavali** [reply]
Did you check the headers?

**Comment by jaidesai-fn** [comment]
@yasithab-pf - https://docs.fluxninja.com/concepts/flow-control/flow-label#classifiers

Its an open source project called Aperture from FluxNinja - https://github.com/fluxninja/aperture
